### ISSUE TITLE
Capturing an instance of SnapshotRenderer

### ISSUE THREAD

**Issue by colepatrickturner** [4Catalyzer/found-relay#48]
I'm currently running into an issue where I'm doing a modal that involves capturing a previous instance of SnapshotRender. 

#### Here is my render method to better understand:
```
        <Main {...mainProps}>
          {isModal ? this.previousChildren : this.children()}
          {isModal && this.children()}
        </Main>
```

Where `this.previousChildren` is the previous route's SnapshotRenderer and `this.children()` is a modal containing this route's SnapshotRenderer.

####  The error I am observing:
[`QuerySubscription already has a listener.`](https://github.com/4Catalyzer/found-relay/blob/next/src/modern/QuerySubscription.js#L91)

####  The navigation I observe this is as follows:
1. /random-slug
- A new QuerySubscription is created
- Subscribe is called via componentDidMount
2. /editor/id-for-random-slug (modal: true)
- A new QuerySubscription is created and subscribe is called via componentDidMount (for new route)
- The old QuerySubscription will resubscribe via componentWillReceiveProps

- /route-a is captured as `previousChildren`

####  Here is my router schema:
```jsx
  <Route
    path="/"
    Component={AppWrapper}
  >

     <Route path=":nodeID" >
    <Route path="editor">
        <Route path="/" Component={NodeContainer} query={NodeQuery} />
        <Route path="version/:versionID" Component={NodeContainer} query={NodeQuery} />
        </Route>
      </Route>
      <Route path=":slug" Component={SlugContainer} query={SlugQuery} />
  </Route>
```

I am wondering if I am doing something wrong by capturing the instance in `previousChildren` or if there is a way to keep rendering a route beneath a modal.

**Comment by taion** [comment]
I hadn't thought about that modal case. What you're reporting sounds a bit odd to me, though. When you say:

> The old QuerySubscription will resubscribe via componentWillReceiveProps

Is that via the active route in the modal, or via the previous route?

**Comment by colepatrickturner** [comment]
The old QuerySubscription in the model above is the previous route from which I navigated. In this case :nodeID.

Updated original comment route schema, `:nodeID` was previously inside `editor` (transcription error). `:nodeID` and `editor` are siblings.

**Comment by taion** [comment]
But it should only resubscribe if `nextProps.querySubscription` is different from `this.props.querySubscription`, no? https://github.com/4Catalyzer/found-relay/blob/e994d99786ed11e47fd8e0b4a435189050e22635/src/modern/ReadyStateRenderer.js#L47-L51

**Comment by colepatrickturner** [comment]
I looked at `relayContext` on both `nextProps.querySubscription` and the `QuerySubscription` from the initial route - same variables. Not sure if that is enough certainty.

**Comment by taion** [comment]
Check if they're the same object. Also check what `operation` looks like on both of them.

**Comment by colepatrickturner** [comment]
Okay it seems both the querySubscription and the operations differ in this instance of `SnapshotRenderer.componentWillReceiveProps`

Here is what I am observing.
*First page impression*
- `/random-slug` creates a new QuerySubscription mounts a SnapshotRenderer for the `/:nodeID` route.
*Navigate to `/editor/:slug`
- A new QuerySubscription is created with the operation for incoming route.
- A new SnapshotRenderer is mounted with the above QuerySubscription
- The previous SnapshotRenderer.componentWillReceiveProps receives a new `querySubscription` which also happens to be the new QuerySubscription. (this doesn't feel right)
 - The operations that are newly mounted and updated coincide with the new route.

**Comment by taion** [comment]
Okay, that sort of makes sense, though I don't fully understand. I'll take a look.

**Comment by colepatrickturner** [comment]
What is the barrier to rendering modals via different routes? Is there an easy way to overcome it?

**Comment by taion** [comment]
Need to fix the code in QuerySubscription to support multiple listeners, then add test coverage to make sure we capture these cases. I've been slammed lately and haven't had time, though.

**Comment by colepatrickturner** [comment]
Okay, are you looking for PRs?

**Comment by taion** [comment]
A test case (within this test suite) would be helpful. I have a basic implementation up on https://github.com/4Catalyzer/found-relay/pull/66 which you can test to see if it would fix your issue.

**Comment by colepatrickturner** [comment]
Awesome, I've tested and it works flawlessly. Submitted a PR to the PR for test coverage. Cheers!

**Comment by taion** [comment]
Released in v0.3.0-alpha.6.
